tools/data-handler/src/show.ts

```diff
@@ -281,9 +281,14 @@ export class Show {
      * @todo: add unit tests
      * @returns all templates in a project.
      */
-    public async showTemplatesWithDetails(projectPath: string): Promise<(resource)[]> {
+    public async showTemplatesWithDetails(projectPath: string): Promise<template[]> {
         Show.project = new Project(projectPath);
-        return await Show.project.templates();
+        const promiseContainer = [];
+        for (const template of await Show.project.templates()) {
+            promiseContainer.push(Show.project.createTemplateObjectByName(template.name).then(t => t?.show()));
+        }
+        const result = await Promise.all(promiseContainer);
+        return result.filter(item => item) as template[];
```

Comment:
This is fine, but a nice trick to filter undefined values from an array is to use `result.filter(Boolean)`.